src/main/java/com/google/sps/utils/QuizTimingPropertiesUtils.java

```diff
@@ -125,4 +125,29 @@ public String getNewQuestion(Entity game_entity, DatastoreService datastore) {
         return (update_game_entity.getProperty("quizQuestion")).toString();
     }
 
+    //Gives the user points if they have taken a quiz
+    public Boolean giveUserPoints(Boolean userQuizStatus, Entity currentUser, DatastoreService datastore) {
+        if(currentUser == null) {
+            log.log(Level.SEVERE, "Given a null {0}", currentUser);
+            return null;
+        }
+
+        if (datastore == null) {
+            log.log(Level.SEVERE, "Given a null {0}", datastore);
+            return null;
+        }
+
+        if(userQuizStatus) {
+            Key user_key = currentUser.getKey();
+            datastore.delete(user_key);
+
+            Entity updated_user = new Entity("user");
+            updated_user.setProperty("quiz_timestamp", System.currentTimeMillis());
+            updated_user.setProperty("userID", currentUser.getProperty("userID"));
+            updated_user.setProperty("score", ((int) currentUser.getProperty("score")) + 20);
+            datastore.put(updated_user);
```

Comment:
This could be generalized and very useful later. If we pull the logic of adding points to a user entity into a helper function (maybe UserUtils) then we can use this in multiple places. For example, we'll want to give points to a user if they are picked in a quiz, so we'll need to use a generalized version of this logic there as well.

Reply:
I think the only thing that would be replicated here is updating the score because the timestamp only needs to be updated this once when involving the user taking the quiz. Also, I just reduced all the code around it

Reply:
yes, a good reason to refactor out updating the score is because what if we want to assign more or less than 20 points for taking a quiz? If we have the number 20 hard coded as a part of this method, then our test cases will all break (as they rely on the number 20 being the exact number). Instead, if we take this method and refactor it out and have it take a generic number of points, we can test that method and never have to worry about the tests breaking. Then, when we test the larger method, we don't need to test that the user's score actually increased because we can assume that calling that function is an already well-tested method.